app/controllers/organisations/new.js

```diff
@@ -0,0 +1,116 @@
+import Ember from "ember";
+
+import SearchOptionMixin from "stock/mixins/search_option";
+import GoodcityController from "../goodcity_controller";
+import AsyncMixin from "stock/mixins/async";
+
+export default GoodcityController.extend(SearchOptionMixin, AsyncMixin, {
+  organisationService: Ember.inject.service(),
+  messageBox: Ember.inject.service(),
+  i18n: Ember.inject.service(),
+
+  showError: false,
+  name_en: "",
+  name_zh_tw: "",
+  website: "",
+  validate: false,
+
+  isInValidNameEn: Ember.computed("name_en", function() {
```

Comment:
I think we can move some of the common methods to mixins with name `organisations_validations` mixins ?
I have seen this common pattern of too much validations in `controller` which is making controlling too much bloated. I think we can create `mixins/validations` and use it in controllers.
Thoughts ?

@patrixr

Reply:
If you create a mixin just for organisations, then it likely won't ever be used anywhere else. Which defeats the purpose of a mixin.

The computed property `isInValidNameEn` with its own specific validation doesn't have a place in a mixin, because its name is custom to this controller, and it's validation logic is custom as well.

You can create a general validation mixin, e.g `mixins/validations` in which you can have helper methods for validations.

e.g maybe something like

```javascript
import ValidationMixin, { notEmpty, gte, isUrl, matches } from '../mixins/validations'

export default Controller.extend(ValidationMixin, {
    validations: {
        name:      [ notEmpty() ],
        website:   [ isUrl(), matches('goodcity.hk') ],
        age:       [ gte(18) ]
    }

   // The mixin code would read the configuration above ☝️  and generate computed properties like `isNameValid`
})
```

Basically --> you can put reusable tools in a mixin, even generated computed properties. But the controller must decide what/how it is structured

Reply:
I'm very much in support of writing reusable tools, but I think it should probably be done outside of the scope of this PR